code/datums/diseases/advance/symptoms/oxygen.dm

```diff
@@ -28,13 +28,16 @@ Bonus
 	symptom_delay_min = 1
 	symptom_delay_max = 1
 	var/regenerate_blood = FALSE
+	var/already_NOBREATH = FALSE
 	threshold_desc = "<b>Resistance 8:</b>Additionally regenerates lost blood.<br>"
 
 /datum/symptom/oxygen/Start(datum/disease/advance/A)
 	if(!..())
 		return
 	if(A.properties["resistance"] >= 8) //blood regeneration
 		regenerate_blood = TRUE
+	if(A.affected_mob.has_trait(TRAIT_NOBREATH)) // with inorganic biology one could already have NOBREATH
+		already_NOBREATH = TRUE

```

Comment:
You're using the trait system wrong here. One of the strengths of traits as opposed to flags is that they track SOURCES of a trait. So you don't need to keep track of the mob already having the trait, you've just to got add (and remove) your trait with its own unique source.

Something like:
// When symptom means you don't need to breath
mob.add_trait(TRAIT_NOBREATH, DISEASE_TRAIT)

// When the disease is cured or the stage goes down
mob.remove_trait(TRAIT_NOBREATH, DISEASE_TRAIT)

You'll need to define DISEASE_TRAIT in code/__DEFINES/traits.dm, near the other sources of traits.

Reply:
It should use the trait system correctly now